### DISCUSSION TITLE
Field array reset not working

### DISCUSSION THREAD

**Discussion by kaisiant** [react-hook-form/react-hook-form#3308] [Q&A] [answered]
I found an existing question on this topic https://github.com/react-hook-form/react-hook-form/issues/1163
but it is not really solving my problem.
So, I modified the CSB: https://codesandbox.io/s/field-array-not-loading-values-from-reset-forked-x4m1b?file=/src/App.js:459-861

The issue I found between these two controls:

1) Reset not working
```
 <Controller
                  name={`array[${index}].value`}
                  as={<input type="text" />}
                  control={control}
                  defaultValue={item.value}
 />
```

2) Reset works fine
```
<input
                  name={`array[${index}].value`}
                  type="text"
                  defaultValue={item.value}
                  ref={register}
 />
```

What am I did wrong?

**Comment by bluebill1049** [comment, marked as answer]
the problem is `id`, we are having a warning in the next update.  id is for keyName with useFieldArray, unless you change to something else with keyName prop
